packages/medusa/src/services/swap.ts

```diff
@@ -1060,61 +1120,78 @@ class SwapService extends BaseService {
           })
 
           if (shippedQty !== i.quantity) {
-            swap.fulfillment_status = "partially_shipped"
+            swap.fulfillment_status = SwapFulfillmentStatus.PARTIALLY_SHIPPED
           }
         } else {
           if (i.shipped_quantity !== i.quantity) {
-            swap.fulfillment_status = "partially_shipped"
+            swap.fulfillment_status = SwapFulfillmentStatus.PARTIALLY_SHIPPED
           }
         }
       }
 
-      const swapRepo = manager.getCustomRepository(this.swapRepository_)
       const result = await swapRepo.save(swap)
+
       await this.eventBus_
         .withTransaction(manager)
         .emit(SwapService.Events.SHIPMENT_CREATED, {
           id: swapId,
           fulfillment_id: shipment.id,
           no_notification: swap.no_notification,
         })
+
       return result
     })
   }
 
   /**
    * Dedicated method to delete metadata for a swap.
-   * @param {string} swapId - the order to delete metadata from.
-   * @param {string} key - key for metadata field
-   * @return {Promise} resolves to the updated result.
+   *
+   * @param swapId - the order to delete metadata from.
+   * @param key - key for metadata field
+   * @return resolves to the updated result.
    */
-  async deleteMetadata(swapId, key) {
-    const validatedId = this.validateId_(swapId)
+  async deleteMetadata(swapId: string, key: string): Promise<Swap | never> {
```

Comment:
question: This one is not use too, should we remove it?

Reply:
Think it is fine to remove yes 👍